src/backup_status.c

```diff
@@ -348,6 +348,32 @@ backup_status_init(backup_status_t* status, backup_config_t* conf)
 		}
 	}
 
+	if (conf->prefer_racks) {
+		as_vector rackids;
+		as_vector_init(&rackids, sizeof(char*), 1);
+
+		char* racks_clone = strdup(conf->prefer_racks);
+		split_string(racks_clone, ',', true, &rackids);
+
+		for (uint32_t i = 0; i < rackids.size; i++) {
+			char* id_str = (char*) as_vector_get_ptr(&rackids, i);
+			int64_t id = 0;
+			if (!better_atoi(id_str, &id) || id < 0 || id > MAX_RACKID) {
+				err("Invalid rack id %s", id_str);
+
+				cf_free(racks_clone);
+				as_vector_destroy(&rackids);
+
+				goto cleanup2;
+			}
+
+			as_config_add_rack_id(&as_conf, (int)id);
+		}
```

Comment:
Do we have a way to test this?

Reply:
I'm not sure how to tell that it is active from a test. I just checked that the value gets set in the C client policy from a debugger.

Reply:
I'll add it to an integration test to exercise that part of the code in that tests.